src/zhinst/toolkit/driver/devices/pqsc.py

```diff
@@ -150,23 +150,24 @@ def check_ref_clock(self, *, timeout: int = 30, sleep_time: int = 1) -> bool:
 
     def check_zsync_connection(
         self,
-        ports: Union[List[int], int] = 0,
+        ports: Union[List[int], int, List[BaseInstrument], BaseInstrument],
         *,
-        timeout: int = 30,
-        sleep_time: int = 1,
+        timeout: float = 10,
+        sleep_time: float = 0.1,
     ) -> Union[List[bool], bool]:
-        """Check if the ZSync connection on the given port(s) is established.
+        """Check if a ZSync connection is established.
 
-        This function checks the current status of the instrument connected to
-        the given ports.
+        Checks the current status of the instrument connected to the given ports.
+        If a instrument(s) is given instead of a port number, first finds the correct
+        port number(s).
 
         Args:
             ports: The port numbers to check the ZSync connection for.
-                It can either be a single port number given as integer or a list
-                of several port numbers. (default: 0)
-            timeout: Maximum time in seconds the program waits (default: 30).
+                It can either be a single port number given as integer, a list
+                of several port numbers an instrument or a list of instruments.
+            timeout: Maximum time in seconds the program waits (default: 10).
             sleep_time: Time in seconds to wait between requesting the reference
-                clock status (default: 1)
+                clock status (default: 0.1)
 
         Raises:
             TimeoutError: If the process of establishing a ZSync connection on
```

Comment:
this changes the interface of the function so we should add a 

.. versionchanged:: 0.6.1

Reply:
done.